src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/EligibilityService.kt

```diff
@@ -14,43 +15,80 @@ infix fun EligibilityCheck.describedAs(message: String): Pair<EligibilityCheck,
+  fun getIneligibilityReasons(prisoner: PrisonerSearchPrisoner): List<String> {
+    val genericIneligibilityReasons = genericChecks.mapNotNull { (test, message) -> if (!test(prisoner)) message else null }
 
-  private fun isPersonParoleEligible(): EligibilityCheck = early@{
-    if (it.paroleEligibilityDate != null) {
-      if (it.paroleEligibilityDate.isAfter(LocalDate.now(clock))) {
-        return@early true
+    val crdIneligibilityReasons = crdChecks.mapNotNull { (test, message) -> if (!test(prisoner)) message else null }
+    // If eligible for CRD licence, return as eligible
+    if (genericIneligibilityReasons.isEmpty() && crdIneligibilityReasons.isEmpty()) {
+      return emptyList()
+    }
+
+    var recallIneligibilityReasons = emptyList<String>()
+    if (recallEnabled) {
+      recallIneligibilityReasons = recallChecks.mapNotNull { (test, message) -> if (!test(prisoner)) message else null }
+      // If eligible for PRRD licence, return as eligible
+      if (genericIneligibilityReasons.isEmpty() && recallIneligibilityReasons.isEmpty()) {
+        return emptyList()
       }
     }
-    return@early false
+
+    // If not eligible for either, return combined list of reasons
+    return (genericIneligibilityReasons + crdIneligibilityReasons + recallIneligibilityReasons).distinct()
   }
 
-  private fun isNotDead(): EligibilityCheck = { it.legalStatus != "DEAD" }
+  // CRD-specific eligibility rules
+  private fun hasConditionalReleaseDate(): EligibilityCheck = { it.conditionalReleaseDate != null }
 
-  private fun isOnIndeterminateSentence(): EligibilityCheck = {
-    if (it.indeterminateSentence == null) {
-      log.warn("${it.prisonerNumber} missing indeterminateSentence")
-    }
-    it.indeterminateSentence ?: false
+  private fun hasCrdTodayOrInTheFuture(): EligibilityCheck = early@{
+    val releaseDate = it.conditionalReleaseDate ?: return@early true
+    releaseDate.isEqual(LocalDate.now(clock)) || releaseDate.isAfter(LocalDate.now(clock))
   }
 
-  private fun hasConditionalReleaseDate(): EligibilityCheck = { it.conditionalReleaseDate != null }
+  private fun isRecallCase(): EligibilityCheck = early@{
+    // If a CRD but no PRRD it should NOT be treated as a recall
+    if (it.conditionalReleaseDate != null && it.postRecallReleaseDate == null) {
+      return@early false
+    }
+
+    if (it.conditionalReleaseDate != null && it.postRecallReleaseDate != null) {
+      // If the PRRD > CRD - it should be treated as a recall otherwise it is not treated as a recall
+      return@early it.postRecallReleaseDate.isAfter(it.conditionalReleaseDate)
+    }
+
+    // Trust the Nomis recall flag as a fallback position - the above rules should always override
+    if (it.recall == null) {
+      log.warn("${it.prisonerNumber} missing recall flag")
+    }
+    return@early it.recall ?: false
```

Comment:
mmh.. we have a couple of ways of determining whether a case is a recall case now - it would make sense if these were unified at some point.

Reply:
While it's marked as new, this is the existing eligibility check that we have
The "core" check seems to follow the same logic we use now, `it.postRecallReleaseDate.isAfter(it.conditionalReleaseDate)`, but agreed that some standardisation would probably be a good thing